src/Nethermind/Nethermind.Core/TypeDiscovery.cs

```diff
@@ -13,14 +13,20 @@ namespace Nethermind.Core
     public static class TypeDiscovery
     {
         private static readonly HashSet<Assembly> _nethermindAssemblies = new();
```

Comment:
Does `_nethermindAssemblies` contain Nethermind assemblies only? Looks like it doesn't. So I'd suggest renaming it.

Reply:
Do you have name suggestion?